tests/Tests.AzureAppConfiguration/MapTests.cs

```diff
@@ -0,0 +1,332 @@
+        value: @"
+                        {
+                            ""uri"":""https://keyvault-theclassics.vault.azure.net/secrets/TheTrialSecret""
+                        }",
+        eTag: new ETag("c3c231fd-39a0-4cb6-3237-4614474b92c1"),
+        contentType: KeyVaultConstants.ContentType + "; charset=utf-8");
+
+        TimeSpan CacheExpirationTime = TimeSpan.FromSeconds(1);
+
+        string _certValue = "Certificate Value from KeyVault";
+        Uri vaultUri = new Uri("https://keyvault-theclassics.vault.azure.net");
+
+        [Fact]
+        public void MapTransformKeyValue()
+        {
+            var mockClient = GetMockConfigurationClient();
+
+            var mockClientManager = TestHelpers.CreateMockedConfigurationClientManager(mockClient.Object);
+
+            var config = new ConfigurationBuilder()
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.ClientManager = mockClientManager;
+                    options.Map((setting) =>
+                    {
+                        if (setting.Key == "TestKey1")
+                        {
+                            setting.Value += " mapped";
+                        }
+                        return new ValueTask<ConfigurationSetting>(setting);
+                    }).Map((setting) =>
+                    {
+                        if (setting.Value.EndsWith("mapped"))
+                        {
+                            setting.Value += " first";
+                        }
+                        else
+                        {
+                            setting.Value += " second";
+                        }
+                        return new ValueTask<ConfigurationSetting>(setting);
+                    });
+                })
+                .Build();
+
+            Assert.Equal("TestValue1 mapped first", config["TestKey1"]);
+            Assert.Equal("TestValue2 second", config["TestKey2"]);
+        }
+
+        [Fact]
+        public void MapTransformKeyVaultValueBeforeAdapters()
+        {
+            var mockClient = GetMockConfigurationClient();
+            mockClient.Setup(c => c.GetConfigurationSettingsAsync(It.IsAny<SettingSelector>(), It.IsAny<CancellationToken>()))
+                .Returns(new MockAsyncPageable(new List<ConfigurationSetting> { _kvr }));
+
+            var mockClientManager = TestHelpers.CreateMockedConfigurationClientManager(mockClient.Object);
+
+            var mockSecretClient = new Mock<SecretClient>(MockBehavior.Strict);
+            mockSecretClient.SetupGet(client => client.VaultUri).Returns(vaultUri);
+            mockSecretClient.Setup(client => client.GetSecretAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<CancellationToken>()))
+                .Returns((string name, string version, CancellationToken cancellationToken) =>
+                    Task.FromResult((Response<KeyVaultSecret>)new MockResponse<KeyVaultSecret>(new KeyVaultSecret(name, _certValue))));
```

Comment:
Would this setup always return `_certValue` for any secret name? If that's the case, this test will pass even if you dont have the mapper changing secret name.